src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      );
+    }
+
+    if (axiosInstance === undefined) {
+      axiosInstance = axios.create();
+    }
+    axiosInstance.defaults.decompress = true;
+    axiosInstance.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        if (NeynarAPIClient.isApiErrorResponse(error)) {
+          const apiErrors = error.response.data;
+          this.logger.warn(`API errors: ${JSON.stringify(apiErrors)}`);
+        }
+        throw error;
+      }
+    );
+
+    const v1Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v1",
+      apiKey: apiKey,
+    });
+    const v2Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v2",
+      apiKey: apiKey,
+    });
+    this.apis = {
+      v1Cast: new CastApiV1(v1Config, undefined, axiosInstance),
+      user: new UserApi(v1Config, undefined, axiosInstance),
+      verification: new VerificationApi(v1Config, undefined, axiosInstance),
+      notifications: new NotificationsApi(v1Config, undefined, axiosInstance),
+      reactions: new ReactionsApi(v1Config, undefined, axiosInstance),
+      v1Follows: new FollowsApiV1(v1Config, undefined, axiosInstance),
+      v2Cast: new CastApiV2(v2Config, undefined, axiosInstance),
+      signer: new SignerApi(v2Config, undefined, axiosInstance),
+      reaction: new ReactionApi(v2Config, undefined, axiosInstance),
+      v2Follow: new FollowApiV2(v2Config, undefined, axiosInstance),
+    };
+  }
+
+  /**
+   * Utility for parsing errors returned by the Merkle API server. Returns true
+   * if the given error is caused by an error response from the server, and
+   * narrows the type of `error` accordingly.
+   */
+  public static isApiErrorResponse(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    error: any
+  ): error is WithRequired<AxiosError<ErrorRes>, "response"> {
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
+  }
+
+  /**
+   * Creates a Signer. See [Neynar documentation](https://docs.neynar.com/reference/create-signer)
+   * for more details.
+   */
+  public async createSigner(): Promise<Signer> {
+    const response = await this.apis.signer.createSigner();
+    return response.data;
+  }
+
+  /**
+   * Fetches an existing Signer. See [Neynar documentation](https://docs.neynar.com/reference/get-signer)
+   * for more details.
+   *
+   */
+  public async fetchSigner(signerUuid: string): Promise<Signer | undefined> {
+    try {
+      const response = await this.apis.signer.signer(signerUuid);
+      return response.data;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Registers a Signer with an fid. See [Neynar documentation](https://docs.neynar.com/reference/register-app-fid)
```

Comment:
Do we want to point to Neynar documentation for all methods? Only specific ones?

Reply:
We should just point to the docs for all of them

Reply:
We should just point to the docs for all of them